lib/types/object/index.js

```diff
@@ -94,8 +95,22 @@ internals.Object = class extends Any {
         const renamed = {};
         for (let i = 0; i < this._inner.renames.length; ++i) {
             const rename = this._inner.renames[i];
+            const matchedTargetKeys = [];
+
+            const fromIsUndefined = rename.isRegExp ?
+                Object.keys(target).filter((targetKey) => {
```

Comment:
Instead of using `Array.filter` here, can we switch this to utilizing a plain old `for...`. We usually try to steer clear of `Array.filter`, `Array.map`, etc. within the critical path of code due to the performance cost of them.

Reply:
Yeah, based on @Marsup's feedback above this whole block will be re-written and we won't need the `filter` anymore anyways :)